### DISCUSSION TITLE
Eslint rule: enforce no import extensions

### DISCUSSION THREAD

**Discussion by samturrell** [netsells/code-standards#455] [Ideas]
Hi all,

This has come up on another project, where there was an inconsistency on imports, and whether or not they should have a file extensions. E.g.:

```js
import '.../../MyComponent/MyComponent.vue';
import '.../../lib/something.js';
```

vs

```js
import '.../../MyComponent/MyComponent';
import '.../../lib/something';
```

The extension is only ever required if a loader is not set up for a specific file type, it cannot be resolved, or if there's ambiguity around (e.g. a directory with both `MyComponent.vue` and `MyComponent.js` or a file that also resolves a folder such as `import 'store';` might be ambiguous with `./store.js` and `./store/index.js`.

What are your thoughts here? There's a linter rule we can use to automate this and remove it from any future discussions on a PR level

**Comment by cannibble** [comment]
I'm inclined to want the extension visible, just makes it easier to decipher what is being imported without knowing about loaders.

**Reply by cj11489** [reply]
Agreed. I like everything to be explicit, plus avoids the potential ambiguity you mentioned @samturrell

**Reply by samturrell** [reply]
> Agreed. I like everything to be explicit, plus avoids the potential ambiguity you mentioned @samturrell

Worth mentioning though that the odds of that ambiguity happening is very rare

**Reply by cj11489** [reply]
Yeah true. Its mainly personal preference

**Comment by BeccaAnderton** [comment]
I agree with @samturrell - I don't see a need to have the extension visible for imports
